frontend/src/components/DataSourceSelector/DataSourceSelectorToggle.tsx

```diff
@@ -0,0 +1,46 @@
+import classNames from 'classnames';
+import React, { FunctionComponent, useEffect } from 'react';
+import { Dropdown, DropdownProps } from 'semantic-ui-react';
+import { SelectEvent } from './DataSourceSelector';
+
+export type SelectedDatasource = 'non-frozen' | 'frozen' | 'all';
+
+interface ComponentProps {
+  className?: string;
+  onSelect: (selectedData: SelectedDatasource) => void;
+  defaultSource?: SelectedDatasource;
+}
+
+const DataSourceSelectorToggle: FunctionComponent<ComponentProps> = ({ ...props }) => {
+  useEffect(() => {
+    if (props.defaultSource && props.defaultSource.length) {
+      props.onSelect(props.defaultSource);
+    }
+  }, [props.defaultSource]);
```

Comment:
If you're watching for changes here, then this cannot be named a "default" value. So if you only care about the initial value, then this should be
```suggestion
  }, []);
```
And if that's true, then given what you're doing in here, this entire `useEffect` wouldn't be required because the default value would be available in the parent component.

Reply:
Removed unnecessary useEffect